marshmallow/marshalling.py

```diff
@@ -263,7 +265,9 @@ def deserialize(self, data, fields_dict, many=False, partial=False,
                     field_name = field_obj.load_from
                     raw_value = data.get(field_obj.load_from, missing)
                 if raw_value is missing:
-                    if partial:
+                    # Ignore missing field if we're allowed to.
+                    if partial is True or\
+                            is_collection(partial) and attr_name in partial:
```

Comment:
We can avoid type-checking here if the default for `partial` is changed to `tuple()`.